data_handler/handlers/loan_states/hashtack_v0/interest_rate.py

```diff
@@ -54,102 +55,98 @@ def _set_events(self, start_block: int, end_block: int) -> None:
             if interest_rate_state.current_block != event["block_number"]:
-                self._add_block_data(blocks_data, interest_rate_state.build_interest_rate_model(HASHSTACK_ID))
+                self._add_block_data(interest_rate_state.build_interest_rate_model(HASHSTACK_ID))
 
             # Get token name. Validate event `key_name` and token name.
             token_name = TOKEN_MAPPING.get(event["data"][0], "")
+            interest_rate_state.current_block = event["block_number"]
+            interest_rate_state.latest_timestamp = event["timestamp"]
             if not token_name or event["key_name"] != "current_apr":
-                interest_rate_state.current_block = event["block_number"]
                 continue
 
             # Set initial timestamp values for the first token event
             if not self.last_block_data or interest_rate_state.token_timestamps[token_name] == 0:
                 interest_rate_state.token_timestamps[token_name] = event["timestamp"]
-                interest_rate_state.current_block = event["block_number"]
                 continue
 
             # Get needed variables
-            interest_rate_state.current_timestamp = event["timestamp"]
             seconds_passed = interest_rate_state.get_seconds_passed(token_name)
             borrow_apr_bps = Decimal(int(event["data"][1], 16))
             supply_apr_bps = Decimal(int(event["data"][2], 16))
 
             # Calculate interest rate for supply and borrow and convert to percents using the formula:
             # (apr * seconds_passed / seconds_in_year) / 10000
-            current_collateral_change = (supply_apr_bps * seconds_passed / SECONDS_IN_YEAR) * percents_decimals_shift
-            current_debt_change = (borrow_apr_bps * seconds_passed / SECONDS_IN_YEAR) * percents_decimals_shift
+            cumulative_collateral_interest_rate_increase = (
+                    (supply_apr_bps * seconds_passed / SECONDS_IN_YEAR) * percents_decimals_shift
+            )
+            cumulative_debt_interest_rate_increase = (
+                    (borrow_apr_bps * seconds_passed / SECONDS_IN_YEAR) * percents_decimals_shift
+            )
             interest_rate_state.update_state_cumulative_data(
-                token_name, event["block_number"], current_collateral_change, current_debt_change
+                token_name, event["block_number"],
+                cumulative_collateral_interest_rate_increase,
+                cumulative_debt_interest_rate_increase,
             )
 
         # Write last block data
-        self._add_block_data(blocks_data, interest_rate_state.build_interest_rate_model(HASHSTACK_ID))
-        return blocks_data
+        self._add_block_data(interest_rate_state.build_interest_rate_model(HASHSTACK_ID))
 
-    def _get_blocks_bounds(self, start_block, end_block, latest_block) -> tuple[int, int]:
+    def _get_blocks_bounds(self, end_block: int, latest_block: int) -> tuple[int, int]:
```

Comment:
```suggestion
    def _get_blocks_bounds(self, start_block: int, latest_block: int) -> tuple[int, int]:
```

.. This feels more natural and readable.